src/main/java/com/google/sps/servlets/Util.java

```diff
@@ -171,15 +172,29 @@ public void associateRefreshToken(String userId, String refreshToken) {
     datastore.put(tokenEntity);
   }
 
-  public void updateDevices(String userId, List<String> deviceIds, String updatesInJson) throws IOException {
+  public List<String> updateDevices(String userId, List<String> deviceIds, String updatesInJson) throws IOException {
     final String accessToken = getAccessToken(userId);
-    for (final String deviceId : deviceIds) {
-      final String updateURL = getUpdateUrl(deviceId);
-      RequestBody body = RequestBody.create(JSON_TYPE, updatesInJson);
-      Request req = new Request.Builder().url(updateURL).put(body).addHeader("Authorization", "Bearer " + accessToken).build();
-      Response updateResponse = client.newCall(req).execute();
-      updateResponse.body().close();
-    }
+    final List<String> failedUpdateDeviceIds = Collections.synchronizedList(new ArrayList<String>());
+    deviceIds
+      .parallelStream()
+      .forEach(
+        deviceId -> {
+          try {
+            updateSingleDevice(accessToken, deviceId, updatesInJson);
+          } catch (IOException e) {
```

Comment:
Even though it's intended to swallow the exception, it's better to log the exception. In case a device update keeps failing, we may debug using the exception log.